src/experiment/evaluation.jl

```diff
@@ -2,42 +2,41 @@ abstract type AbstractEvaluator end
 
 mutable struct SameInstancesEvaluator <: AbstractEvaluator
     instances::Union{Array{CPModel}, Nothing}
-    eval_freq::Int
-    nb_instances::Int
+    metrics::Union{Array{Any,2}, Nothing} 
+    eval_freq::Int64
+    nb_instances::Int64
+    nb_heuristic::Union{Int64, Nothing}
 end
 
-function SameInstancesEvaluator(; eval_freq = 50, nb_instances = 50)
-    SameInstancesEvaluator(nothing, eval_freq, nb_instances)
-end
-
-function init_evaluator!(eval::SameInstancesEvaluator, generator::AbstractModelGenerator; seed=nothing)
-    instances = Array{CPModel}(undef, eval.nb_instances)
-    for i in 1:eval.nb_instances
+function SameInstancesEvaluator(valueSelectionArray::Array{H, 1}, generator::AbstractModelGenerator; seed=nothing, eval_freq = 50, nb_instances = 10) where H<: ValueSelection
+    instances = Array{CPModel}(undef, nb_instances)
+    metrics = Array{Any,2}(undef,nb_instances, size(valueSelectionArray,1)) 

```

Comment:
In fact, it doesn't work because this kind of declaration doesn't allow the Array to contain basicmetrics with different signature.
As long as the first one is set, let's say basicmetrics{TakeObjective,learnedheuristic{SR, R, A}}, all the metrics in the array as to be of the same type.